rust/connlib/tunnel/src/client.rs

```diff
@@ -1102,6 +1102,15 @@ impl ClientState {
 
         self.set_dns_mapping(dns_mapping);
 
+        self.buffered_events
+            .push_back(ClientEvent::DnsServersChanged {
+                dns_by_sentinel: self
+                    .dns_mapping
+                    .iter()
+                    .map(|(sentinel_dns, effective_dns)| (*sentinel_dns, effective_dns.address()))
+                    .collect(),
```

Comment:
can we use a `BiMap` here?  then we can just use `self.dns_mapping()` and it actually represents a 1:1 mapping

Reply:
We can use a `BiMap` but I want to avoid exposing internal types like `DnsServer` in the API here.